### ISSUE TITLE
Wrong return type for macros named `error`

### ISSUE THREAD

**Issue by FelixStridsberg** [intellij-rust/intellij-rust#7260]
## Environment

* **IntelliJ Rust plugin version: 0.4.147.3871-211**
* **Rust toolchain version: 1.52.1 stable-x86_64-unknown-linux-gnu**
* **IDE name and version: CLion 2021.1.1**
* **Operating system: Ubuntu 20.04.2 LTS**

## Problem description
Naming a macro `error` causes the return type of that macro to be incorrectly inferred as `()`.
The program compiles correctly with a macro named `error`, and clippy does not complain about it, so it seems to be valid rust.

## Steps to reproduce
Paste this in the IDE of the version specified above:
```rust
enum Data {
    Value
}

macro_rules! error {
    ($expr:expr) => {
        Data::Value
    }
}

macro_rules! other_name {
    ($expr:expr) => {
        Data::Value
    }
}

#[allow(unused)]
fn main() {
    let a = other_name!(1 + 1); // Inlay hint shows correctly `a : Data`
    let a = error!(1 + 1); // Inlay hint shows incorrectly `a : ()`

    let a: Data = other_name!(1 + 1); // Shows no errors
    let a: Data = error!(1 + 1); // Shows error: mismatched type [E0308], expected `Data` found `()`
}
```

It compiles fine without any errors or warnings, but the IDE show error.
Expanding the `error` macro via the context actions (CTRL + A) shows a correct macro expansion.

![error-macro](https://user-images.githubusercontent.com/1368168/119705695-5a393f00-be59-11eb-9c45-60d4a135f0b8.jpg)

**Comment by Kobzol** [comment]
The plugin parses `error`, `warn`, `info` etc. macro calls as logging macros, and type inference hard-codes their return type as `()`, without actually resolving them.

@vlad20012 Would it be OK to resolve the macros first in `inferMacroExprType0` (`TypeInferenceWalker:1187`) to see if they are indeed from `log` and if not just treat them as normal macros?

**Comment by vlad20012** [comment]
> Would it be OK to resolve the macros first 

Yes, sure.

**Comment by kumbayo** [comment]
I think this should be solved with the changes made by @Kobzol in #7358.
I could not reproduce the error with the current source code from Git.

**Comment by Kobzol** [comment]
Oh, right, this was actually fixed even sooner in https://github.com/intellij-rust/intellij-rust/pull/7267, but I forgot to mark this issue :sweat_smile: Thanks for reminding me!
